### PR TITLE
Black on SeqIO and AlignIO

### PR DESCRIPTION
**Pull request by peterjc**
This pull request addresses issue #2008

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
These changes look OK to me by eye, what do you think @svalqui or @MarkusPiotrowski ?

File: Bio/AlignIO/ClustalIO.py [hunk 1]
```diff
@@ -165,19 +167,22 @@ def __next__(self):
...
+                            "Could not parse line, bad sequence number:\n%s" % line
+                        )
                     if len(fields[1].replace("-", "")) != letters:
-                        raise ValueError("Could not parse line, "
-                                         "invalid sequence number:\n%s" % line)
+                        raise ValueError(
+                            "Could not parse line, "
+                            "invalid sequence number:\n%s" % line
+                        )
             elif line[0] == " ":
```
Comment (svalqui) [inline Bio/AlignIO/ClustalIO.py]: The Strings above could be joined in one line.

File: Bio/AlignIO/ClustalIO.py [hunk 2]
```diff
@@ -228,13 +233,13 @@ def __next__(self):
                     raise ValueError("Could not parse line:\n%s" % repr(line))
 
                 if fields[0] != ids[i]:
-                    raise ValueError("Identifiers out of order? "
-                                     "Got '%s' but expected '%s'"
-                                     % (fields[0], ids[i]))
+                    raise ValueError(
+                        "Identifiers out of order? "
+                        "Got '%s' but expected '%s'" % (fields[0], ids[i])
+                    )
```
Comment (svalqui) [inline Bio/AlignIO/ClustalIO.py]: The Strings above could be joined in one line.

File: Bio/AlignIO/ClustalIO.py [hunk 3]
```diff
@@ -250,12 +255,14 @@ def __next__(self):
...
+                        raise ValueError(
+                            "Could not parse line, bad sequence number:\n%s" % line
+                        )
                     if len(seqs[i].replace("-", "")) != letters:
-                        raise ValueError("Could not parse line, "
-                                         "invalid sequence number:\n%s" % line)
+                        raise ValueError(
+                            "Could not parse line, "
+                            "invalid sequence number:\n%s" % line
+                        )
```
Comment (svalqui) [inline Bio/AlignIO/ClustalIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/ClustalIO.py [hunk 4]
```diff
@@ -265,23 +272,28 @@ def __next__(self):
...
-                   for (i, s) in zip(ids, seqs))
+        if (
+            self.records_per_alignment is not None
+            and self.records_per_alignment != len(ids)
+        ):
+            raise ValueError(
+                "Found %i records in this alignment, "
+                "told to expect %i" % (len(ids), self.records_per_alignment)
+            )
+
```
Comment (svalqui) [inline Bio/AlignIO/ClustalIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/MafIO.py [hunk 5]
```diff
@@ -625,12 +696,16 @@ def get_spliced(self, starts, ends, strand=1):
...
                         elif ref_first_strand != seqrec.annotations["strand"]:
-                            raise ValueError("Encountered strand='%s' on target seqname, "
-                                             "expected '%s'" %
-                                             (seqrec.annotations["strand"], ref_first_strand))
+                            raise ValueError(
+                                "Encountered strand='%s' on target seqname, "
+                                "expected '%s'"
+                                % (seqrec.annotations["strand"], ref_first_strand)
+                            )
                     except KeyError:
```
Comment (svalqui) [inline Bio/AlignIO/MafIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/NexusIO.py [hunk 6]
```diff
@@ -113,17 +115,19 @@ def write_alignment(self, alignment, interleave=None):
...
             raise ValueError("Non-empty sequences are required")
-        minimal_record = "#NEXUS\nbegin data; dimensions ntax=0 nchar=0; " \
-                         + "format datatype=%s; end;"  \
-                         % self._classify_alphabet_for_nexus(alignment._alphabet)
+        minimal_record = (
+            "#NEXUS\nbegin data; dimensions ntax=0 nchar=0; "
+            + "format datatype=%s; end;"
+            % self._classify_alphabet_for_nexus(alignment._alphabet)
+        )
         n = Nexus.Nexus(minimal_record)
```
Comment (svalqui) [inline Bio/AlignIO/NexusIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/PhylipIO.py [hunk 7]
```diff
@@ -100,9 +100,10 @@ def write_alignment(self, alignment, id_width=_PHYLIP_ID_WIDTH):
...
             name = sanitize_name(record.id, id_width)
             if name in names:
-                raise ValueError("Repeated name %r (originally %r), "
-                                 "possibly due to truncation"
-                                 % (name, record.id))
+                raise ValueError(
+                    "Repeated name %r (originally %r), "
+                    "possibly due to truncation" % (name, record.id)
+                )
             names.append(name)
```
Comment (svalqui) [inline Bio/AlignIO/PhylipIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/PhylipIO.py [hunk 8]
```diff
@@ -211,11 +212,14 @@ def __next__(self):
...
-                             "told to expect %i"
-                             % (number_of_seqs, self.records_per_alignment))
+        if (
+            self.records_per_alignment is not None
+            and self.records_per_alignment != number_of_seqs
+        ):
+            raise ValueError(
+                "Found %i records in this alignment, "
+                "told to expect %i" % (number_of_seqs, self.records_per_alignment)
+            )
```
Comment (svalqui) [inline Bio/AlignIO/PhylipIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/PhylipIO.py [hunk 9]
```diff
@@ -326,9 +330,10 @@ def write_alignment(self, alignment, id_width=_PHYLIP_ID_WIDTH):
...
             name = sanitize_name(record.id, id_width)
             if name in names:
-                raise ValueError("Repeated name %r (originally %r), "
-                                 "possibly due to truncation"
-                                 % (name, record.id))
+                raise ValueError(
+                    "Repeated name %r (originally %r), "
+                    "possibly due to truncation" % (name, record.id)
+                )
             names.append(name)
```
Comment (svalqui) [inline Bio/AlignIO/PhylipIO.py]: Same here, the Strings above could be joined in one line.

File: Bio/AlignIO/PhylipIO.py [hunk 10]
```diff
@@ -387,11 +392,14 @@ def __next__(self):
...
-                             "told to expect %i"
-                             % (number_of_seqs, self.records_per_alignment))
+        if (
+            self.records_per_alignment is not None
+            and self.records_per_alignment != number_of_seqs
+        ):
+            raise ValueError(
+                "Found %i records in this alignment, "
+                "told to expect %i" % (number_of_seqs, self.records_per_alignment)
+            )
```
Comment (svalqui) [inline Bio/AlignIO/PhylipIO.py]: Same here, the Strings above could be joined in one line.

**Comment by svalqui** [comment]
I have added few comments.

File: Bio/AlignIO/MafIO.py (hunk 5 above)
Comment (peterjc) [reply Bio/AlignIO/MafIO.py]: This line would then be over 88 characters long.

File: Bio/AlignIO/NexusIO.py (hunk 6 above)
Comment (peterjc) [reply Bio/AlignIO/NexusIO.py]: Also we don't need the plus sign.

**Comment by peterjc** [comment]
Thank you! I did wonder about a couple of these - you're right apart from ``Bio/AlignIO/MafIO.py`` which would then exceed 88 characters. Updating...

**Pull request merged**